src/main/java/me/moodcat/api/RoomAPI.java

```diff
@@ -56,15 +58,11 @@ public RoomAPI(final RoomDAO roomDAO, final ChatDAO chatDAO) {
         VAVector targetVector = VAVector.createTargetVector(moods);
         List<Room> allRooms = roomDAO.listRooms(targetVector, limit);
 
-        if (limit > allRooms.size()) {
+        if (limit > allRooms.size() || limit < 1) {
```

Comment:
I would rather see a limit of `null` to be used. A limit of `0` is confusing, I would expect `0` results then, and not all.